src/revenue.js

```diff
@@ -0,0 +1,158 @@
+  if (type(quantity) !== 'number') {
+    utils.log('Unsupported type for quantity: ' + type(quantity) + ', expecting number');
+  } else {
+    this._quantity = parseInt(quantity);
+  }
+  return this;
+};
+
+/**
+ * Set a value for the price. This field is required for all revenue being logged.
+ * Note revenue amount is calculated as price * quantity.
+ * @public
+ * @param {number} price - Double value for the quantity.
+ * @return {Revenue} Returns the same Revenue object, allowing you to chain multiple method calls together.
+ * @example var revenue = new amplitude.Revenue().setProductId('productIdentifier').setPrice(10.99);
+ * amplitude.logRevenueV2(revenue);
+ */
+Revenue.prototype.setPrice = function setPrice(price) {
+  if (type(price) !== 'number') {
+    utils.log('Unsupported type for price: ' + type(price) + ', expecting number');
+  } else {
+    this._price = price;
+  }
+  return this;
+};
+
+/**
+ * Set a value for the revenueType (for example purchase, cost, tax, refund, etc).
+ * @public
+ * @param {string} revenueType - RevenueType to designate.
+ * @return {Revenue} Returns the same Revenue object, allowing you to chain multiple method calls together.
+ * @example var revenue = new amplitude.Revenue().setProductId('productIdentifier').setPrice(10.99).setRevenueType('purchase');
+ * amplitude.logRevenueV2(revenue);
+ */
+Revenue.prototype.setRevenueType = function setRevenueType(revenueType) {
+  if (type(revenueType) !== 'string') {
+    utils.log('Unsupported type for revenueType: ' + type(revenueType) + ', expecting string');
+  } else {
+    this._revenueType = revenueType;
+  }
+  return this;
+};
+
+/**
+ * Set event properties for the revenue event.
+ * @public
+ * @param {object} properties - Revenue event properties to set.
+ * @return {Revenue} Returns the same Revenue object, allowing you to chain multiple method calls together.
+ * @example var event_properties = {'city': 'San Francisco'};
+ * var revenue = new amplitude.Revenue().setProductId('productIdentifier').setPrice(10.99).setRevenueProperties(event_properties);
+ * amplitude.logRevenueV2(revenue);
+*/
+Revenue.prototype.setRevenueProperties = function setRevenueProperties(revenueProperties) {
+  if (type(revenueProperties) !== 'object') {
+    utils.log('Unsupported type for revenueProperties: ' + type(revenueProperties) + ', expecting object');
+  } else {
+    this._properties = utils.validateProperties(revenueProperties);
+  }
+  return this;
+};
+
+/**
+ * @private
+ */
+Revenue.prototype._isValidRevenue = function _isValidRevenue() {
+  if (type(this._productId) !== 'string' || utils.isEmptyString(this._productId)) {
+    utils.log('Invalid revenue, need to set productId field');
+    return false;
+  }
+  if (type(this._price) !== 'number') {
+    utils.log('Invalid revenue, need to set price field');
+    return false;
+  }
+  return true;
+};
+
+/**
+ * @private
+ */
+Revenue.prototype._toJSONObject = function _toJSONObject() {
```

Comment:
Our event property validator helper function prints warnings when it sees null values, so we only set values if they are not null to prevent the warning from printing every time